vector3/vector3.go

```diff
@@ -515,8 +540,28 @@ func (v Vector[T]) Sub(other Vector[T]) Vector[T] {
 	}
 }
 
-func (v Vector[T]) Dot(other Vector[T]) float64 {
-	return float64((v.x * other.x) + (v.y * other.y) + (v.z * other.z))
+func (v Vector[T]) ReciprocalF() Vector[float32] {
+	return Vector[float32]{
+		x: 1.0 / float32(v.x),
+		y: 1.0 / float32(v.y),
+		z: 1.0 / float32(v.z),
+	}
+}
+
+func (v Vector[T]) Reciprocal() Vector[float64] {
+	return Vector[float64]{
+		x: 1.0 / float64(v.x),
+		y: 1.0 / float64(v.y),
+		z: 1.0 / float64(v.z),
+	}
+}
+
+func (v Vector[T]) Product() T {
```

Comment:
I know this is silly, but I'm having trouble finding a link to what this operation is. Could you provide a link to a math article talking about a vector's self product?

Reply:
That one is used usually when you store a 2d array in 1d array. For example instead of MapTile[10][20] you have MapTile[10*20]

Reply:
Gotcha! What would you think about renaming this to the proper geometric term for each dimension?

2D - Area
3D - Volume
4D - HyperVolume .... okay that one's not that great. But it's technically correct!

Reply:
Also how does that actually work for the 1D array stuff? Wouldn't you have a collision? IE: vector2.New(10, 20) and vector2.New(20, 10) collides and will take up the same space

Reply:
Yes, that was that Unwrap function where magic happens. When you convert index back to coordinate you do this
x = i%size.x
y = i/size.x
and restore x,y from one dimensional index.

>2D - Area
>3D - Volume
>4D - HyperVolume .... okay that one's not that great. But it's technically correct!

Idk, that kinda feels wrong. Not the right place for this names - they describe some body dimensions. Also different function names make this objects implement different interfaces. So you will not be theoretically be able to change dimensionality of your code without changing the function being called.

Reply:
> Yes, that was that Unwrap function where magic happens. 

Interesting! Could you add this thought process to the documentation of the Unwrap function? Potentially as an example test?

https://go.dev/blog/examples

> Also different function names make this objects implement different interfaces

Fair point! In my googling I'm actually surprised to find that a better term doesn't exist. I think the most correct term is [Lebesgue measure](https://en.wikipedia.org/wiki/Lebesgue_measure). But god what a terrible name for a function. It also appears the people are okay with simply going by "Volume" for even 2 dimensions. Pretty disappointing there doesn't exist a better term :/ I'm fine with either `Product` or `Volume`. Your choice

https://math.stackexchange.com/questions/1974518/is-there-a-term-for-the-generalization-of-length-area-volume-etc-elements
https://math.stackexchange.com/questions/1608228/what-is-the-general-term-for-concepts-like-length-area-and-volume